src/js/angular/clustermanagement/directives/cluster-configuration/multi-region.directive.js

```diff
@@ -15,7 +23,197 @@ function MultiRegion() {
+                        $scope.tag = '';
+                        $scope.title = config.title;
+                        $scope.onClick = ($event) => $event.stopPropagation();
+                        $scope.ok = ($event) => {
+                            $uibModalInstance.close({
+                                primaryNode: $scope.rpcAddress,
+                                tag: $scope.tag
+                            });
+                        };
+                        $scope.cancel = function($event) {
+                            $uibModalInstance.dismiss('cancel');
+                        };
+                    },
+                    size: 'lg',
+                    warning: true,
+                    backdrop: 'static'
+                }).result)
+                    .then((payload) => {
+                        $scope.setLoader(true);
+                        return ClusterRestService.enableSecondaryMode(payload);
+                    })
+                    .then(() => toastr.success($translate.instant('cluster_management.cluster_configuration_multi_region.secondary_enabled')))
+                    .catch((response) => {
+                        if (response === 'cancel') {
+                            return;
+                        }
+                        const msg = getError(response);
+                        toastr.error(msg, $translate.instant('cluster_management.cluster_configuration_multi_region.error.secondary'));
+                    })
+                    .finally(() => $scope.setLoader(false));
+            };
+
+            $scope.disableSecondaryMode = () => {
+                $scope.setLoader(true);
+                ModalService.openSimpleModal({
+                    title: $translate.instant('cluster_management.cluster_configuration_multi_region.confirm.disable_secondary_mode'),
+                    message: $translate.instant('cluster_management.cluster_configuration_multi_region.confirm.disable_secondary_mode_warning'),
+                    warning: true,
+                    backdrop: 'static',
+                    stopPropagation: true
+                }).result
+                    .then(() => {
+                        $scope.setLoader(true);
+                        return ClusterRestService.disableSecondaryMode();
+                    })
+                    .then(() => toastr.success($translate.instant('cluster_management.cluster_configuration_multi_region.disabled_secondary_mode')))
+                    .catch((response) => {
+                        if (response === 'cancel') {
+                            return;
+                        }
+                        const msg = getError(response);
+                        toastr.error(msg, $translate.instant('cluster_management.cluster_configuration_multi_region.error.disabling'));
+                    })
+                    .finally(() => $scope.setLoader(false));
+            };
+
+            $scope.setLoader = (loader, message) => {
```

Comment:
If you don't call this from the template, then remove it from the $scope and make it private

Reply:
done